src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,20 +38,32 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String latString = ""; 
+    String lngString = "";
+    String product = "";
+    
     try {
-    String latString = request.getParameter("lat");
-    String lngString = request.getParameter("lng");
-    double lat = convertToDouble(latString);
-    double lng = convertToDouble(lngString);
-    MapLocation userLocation = new MapLocation(lat, lng);
-    smallCityService.createUserServiceWithGeolocation(userLocation);
-    } catch(NullPointerException e) {
-      String zip = request.getParameter("zipCode");
-      smallCityService.createUserServiceWithZip(zip);
+      latString = request.getParameter("lat");
+      lngString = request.getParameter("lng");
+      product = request.getParameter("product");
+    } catch (NullPointerException e) {
       LOGGER.warning(e.getMessage() 
            + "Unable to geolocate user, zipCode entered instead.");
     }
-  
+      
+    SearchObject searchObject = new SearchObject(product);
+    if (latString != null && !latString.isEmpty() 
+          && lngString != null && !lngString.isEmpty()) {
+      double lat = convertToDouble(latString);
+      double lng = convertToDouble(lngString);
```

Comment:
The guava library has a nifty method that simplifies this, but this also works. 
If you're curious: https://guava.dev/releases/19.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String)

Reply:
Interesting! Decided to use this instead